clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/KubernetesV2ProviderSynchronizable.java

```diff
@@ -0,0 +1,195 @@
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials;
+import com.netflix.spinnaker.clouddriver.security.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class KubernetesV2ProviderSynchronizable implements CredentialsInitializerSynchronizable {
+
+  private KubernetesV2Provider kubernetesV2Provider;
+  private AccountCredentialsRepository accountCredentialsRepository;
+  private KubernetesV2CachingAgentDispatcher kubernetesV2CachingAgentDispatcher;
+  private KubernetesResourcePropertyRegistry kubernetesResourcePropertyRegistry;
+  private KubernetesConfigurationProperties kubernetesConfigurationProperties;
+  private KubernetesNamedAccountCredentials.CredentialFactory credentialFactory;
+  private KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap;
+  private CatsModule catsModule;
+
+  public KubernetesV2ProviderSynchronizable(
+      KubernetesV2Provider kubernetesV2Provider,
+      AccountCredentialsRepository accountCredentialsRepository,
+      KubernetesV2CachingAgentDispatcher kubernetesV2CachingAgentDispatcher,
+      KubernetesResourcePropertyRegistry kubernetesResourcePropertyRegistry,
+      KubernetesConfigurationProperties kubernetesConfigurationProperties,
+      KubernetesNamedAccountCredentials.CredentialFactory credentialFactory,
+      KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap,
+      CatsModule catsModule) {
+    this.kubernetesV2Provider = kubernetesV2Provider;
+    this.accountCredentialsRepository = accountCredentialsRepository;
+    this.kubernetesV2CachingAgentDispatcher = kubernetesV2CachingAgentDispatcher;
+    this.kubernetesResourcePropertyRegistry = kubernetesResourcePropertyRegistry;
+    this.kubernetesConfigurationProperties = kubernetesConfigurationProperties;
+    this.credentialFactory = credentialFactory;
+    this.kubernetesSpinnakerKindMap = kubernetesSpinnakerKindMap;
+    this.catsModule = catsModule;
+
+    ScheduledExecutorService poller =
+        Executors.newSingleThreadScheduledExecutor(
+            new NamedThreadFactory(KubernetesV2ProviderSynchronizable.class.getSimpleName()));
+  }
+
+  @Override
+  @PostConstruct
+  public void synchronize() {
+    List<String> newAndChangedAccounts = synchronizeAccountCredentials();
+
+    // we only want to initialize caching agents for new or updated accounts
+    Set<KubernetesNamedAccountCredentials> allAccounts =
+        ProviderUtils.buildThreadSafeSetOfAccounts(
+                accountCredentialsRepository,
+                KubernetesNamedAccountCredentials.class,
+                ProviderVersion.v2)
+            .stream()
+            .filter(account -> newAndChangedAccounts.contains(account.getName()))
```

Comment:
Would be more efficient if `newAndChangedAccounts` were a `Set`; currently is `O(M*N)` where `M` is total accounts and `N` is changed accounts.